abstractions/go/api_error.go

```diff
@@ -0,0 +1,21 @@
+package abstractions
+
+import "fmt"
+
+// ApiError is the parent type for errors thrown by the client when receiving failed responses to its requests
+type ApiError struct {
+	Message string
+}
+
+func (e *ApiError) Error() string {
+	if len(e.Message) > 0 {
+		return fmt.Sprint(e.Message)
+	} else {
+		return "error status code received from the API"
+	}
+}
+
+// NewApiError creates a new ApiError instance
+func NewApiError() *ApiError {
```

Comment:
ApiError can be a value object, why pass it by reference?

Reply:
Error is an interface, so for all the type assertions we need a reference anyway.
Also to align with the models which are returning references in their constructors.

Reply:
It's possible to have a value object implement an interface. Returing references is generally discouraged so I wouldn't do it just for alignment.

Reply:
I understand that, allow me to precise my point:

As far as I understand, if a method prototype is `func (*a adapter) SendAsync() (Parsable, error)` (omitted parameters here), because `error` is an interface, if I want to return a value of the ApiError struct as the error (second return value), I need to return a reference to it, not a copy of the value.